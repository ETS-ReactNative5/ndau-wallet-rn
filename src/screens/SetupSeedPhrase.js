import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  ScrollView,
  Text,
  ProgressViewIOS,
  Platform,
  ProgressBarAndroid,
  SafeAreaView,
  NativeModules
} from 'react-native';
import groupIntoRows from '../helpers/groupIntoRows';
import CommonButton from '../components/CommonButton';

const ROW_LENGTH = 3; // 3 items per row

class SetupSeedPhrase extends Component {
  constructor(props) {
    super(props);
    this.state = {
      seedPhrase: []
    };
  }

  componentDidMount = () => {
    this.props.navigator.setStyle({
      drawUnderTabBar: true,
      tabBarHidden: true
    });
    this.generateSeedPhrase();
  };

  generateSeedPhrase = async () => {
    const KeyaddrManager = NativeModules.KeyaddrManager;
    const seeds = await KeyaddrManager.KeyaddrWordsFromBytes('en', this.props.entropy);
    console.debug(`keyaddr's seed words are: ${seeds}`);
    const seedPhrase = seeds.split(/\s+/g);
    this.setState({ seedPhrase: seedPhrase });

    // Shuffle the deck with a Fisher-Yates shuffle algorithm;
    // walks through the array backwards once, exchanging each value
    // with a random element from the remainder of the array
    let map = seedPhrase.map((e, i) => i);
    let arr = seedPhrase.slice();
    for (let i = arr.length - 1; i >= 0; i--) {
      let r = Math.floor(Math.random() * i);
      [ map[i], map[r] ] = [ map[r], map[i] ];
      [ arr[i], arr[r] ] = [ arr[r], arr[i] ];
    }
    this.shuffledWords = arr;
    // turn array inside out
    this.shuffleMap = map.reduce((a, c, i) => {
      a[c] = i;
      return a;
    }, []);
  };

  onPushAnother = () => {
    this.props.navigator.push({
      label: 'SetupConfirmSeedPhrase',
      screen: 'ndau.SetupConfirmSeedPhrase',
      passProps: {
        encryptionPassword: this.props.encryptionPassword,
        userId: this.props.userId,
        qrToken: this.props.qrToken,
        parentStyles: this.props.parentStyles,
        entropy: this.props.entropy,
        seedPhraseArray: this.state.seedPhrase,
        iconsMap: this.props.iconsMap,
        numberOfAccounts: this.props.numberOfAccounts,
        shuffledWords: this.shuffledWords,
        shuffleMap: this.shuffleMap
      },
      navigatorStyle: {
        drawUnderTabBar: true,
        tabBarHidden: true,
        disabledBackGesture: true
      },
      backButtonHidden: true
    });
  };

  render() {
    // chop the words into ROW_LENGTH-tuples
    const words = groupIntoRows(this.state.seedPhrase, ROW_LENGTH);

    let count = 1;
    return (
      <SafeAreaView style={styles.safeContainer}>
        <View style={styles.container}>
          <ScrollView style={styles.contentContainer}>
            <View>
              <Text style={this.props.parentStyles.wizardText}>Seed phrase</Text>
            </View>
            <View>
              {Platform.OS === 'android' ? (
                <ProgressBarAndroid
                  styleAttr="Horizontal"
                  progress={0.625}
                  style={this.props.parentStyles.progress}
                  indeterminate={false}
                />
              ) : (
                <ProgressViewIOS progress={0.625} style={this.props.parentStyles.progress} />
              )}
            </View>
            <View>
              <Text style={this.props.parentStyles.wizardText}>
                Write this phrase down. You will want to store it in a secure location.
              </Text>
            </View>
            {words.map((row, rowIndex) => {
              return (
                <View key={rowIndex} style={styles.rowView}>
                  {row.map((item, index) => {
                    return (
                      <View key={index} style={styles.rowTextView}>
                        <Text
                          style={{
                            color: '#ffffff',
                            fontSize: 20,
                            textAlign: 'center'
                          }}
                        >
                          {count++}. {item}
                        </Text>
                      </View>
                    );
                  })}
                </View>
              );
            })}
          </ScrollView>
          <View style={styles.footer}>
            <CommonButton onPress={this.onPushAnother} title="I wrote it down" />
          </View>
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  safeContainer: {
    flex: 1,
    backgroundColor: '#1c2227'
  },
  container: {
    flex: 1,
    paddingLeft: 10,
    paddingTop: 10,
    paddingRight: 10,
    paddingBottom: 10,

    backgroundColor: '#1c2227'
  },

  contentContainer: {
    flex: 1 // pushes the footer to the end of the screen
  },
  footer: {
    justifyContent: 'flex-end'
  },
  progress: {
    paddingTop: 30,
    paddingBottom: 30
  },
  rowView: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-evenly'
  },
  rowTextView: {
    height: 60,
    width: 100,
    marginBottom: 10,
    marginTop: 10
  }
});

export default SetupSeedPhrase;
