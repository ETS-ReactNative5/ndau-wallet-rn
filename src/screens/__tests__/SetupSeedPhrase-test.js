import React from 'react';
import { StyleSheet, NativeModules } from 'react-native';
import SetupSeedPhrase from '../SetupSeedPhrase';
import { shallow } from 'enzyme';
import sinon from 'sinon';

jest.mock('NativeModules', () => {
  return {
    KeyaddrManager: {
      KeyaddrWordsFromBytes: jest.fn()
    },
    KeyboardObserver: {}
  };
});

describe('testing SetupSeedPhrase...', () => {
  let styles = StyleSheet.create({
    wizardText: {
      color: '#ffffff',
      fontSize: 20
    }
  });
  const navigator = {
    setStyle: () => {}
  };

  beforeEach(() => {});

  const wrapper = shallow(<SetupSeedPhrase navigator={navigator} parentStyles={styles} />);
  const render = wrapper.dive();

  it('renders correctly', () => {
    expect(render).toMatchSnapshot();
  });

  const KeyaddrWordsFromBytes = sinon.spy(NativeModules.KeyaddrManager, 'KeyaddrWordsFromBytes');

  it('calls KeyaddrWordsFromBytes as expected', () => {
    const wrapper = shallow(<SetupSeedPhrase navigator={navigator} parentStyles={styles} />);
    wrapper.dive();
    expect(KeyaddrWordsFromBytes.calledOnce).toBe(true);
  });
});
