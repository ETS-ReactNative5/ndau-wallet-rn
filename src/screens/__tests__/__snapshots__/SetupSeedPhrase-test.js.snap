// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SetupSeedPhrase... renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SafeAreaView
    style={
      Object {
        "backgroundColor": "#1c2227",
        "flex": 1,
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "#1c2227",
          "flex": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        }
      }
    >
      <ScrollView
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Component>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Seed phrase
          </Text>
        </Component>
        <Component>
          <ProgressViewIOS
            progress={0.625}
          />
        </Component>
        <Component>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Write this phrase down. You will want to store it in a secure location.
          </Text>
        </Component>
        <Component
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-evenly",
            }
          }
        />
        <Component
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-evenly",
            }
          }
        />
        <Component
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-evenly",
            }
          }
        />
        <Component
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-evenly",
            }
          }
        />
      </ScrollView>
      <Component
        style={
          Object {
            "justifyContent": "flex-end",
          }
        }
      >
        <Button
          color="#4d9678"
          onPress={[Function]}
          title="I wrote it down"
        />
      </Component>
    </Component>
  </SafeAreaView>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component
        style={
          Object {
            "backgroundColor": "#1c2227",
            "flex": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
      >
        <ScrollView
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Component>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                }
              }
            >
              Seed phrase
            </Text>
          </Component>
          <Component>
            <ProgressViewIOS
              progress={0.625}
            />
          </Component>
          <Component>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#ffffff",
                  "fontSize": 20,
                }
              }
            >
              Write this phrase down. You will want to store it in a secure location.
            </Text>
          </Component>
          <Component
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-evenly",
              }
            }
          />
          <Component
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-evenly",
              }
            }
          />
          <Component
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-evenly",
              }
            }
          />
          <Component
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-evenly",
              }
            }
          />
        </ScrollView>
        <Component
          style={
            Object {
              "justifyContent": "flex-end",
            }
          }
        >
          <Button
            color="#4d9678"
            onPress={[Function]}
            title="I wrote it down"
          />
        </Component>
      </Component>,
      "style": Object {
        "backgroundColor": "#1c2227",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ScrollView
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Component>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 20,
                  }
                }
              >
                Seed phrase
              </Text>
            </Component>
            <Component>
              <ProgressViewIOS
                progress={0.625}
              />
            </Component>
            <Component>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 20,
                  }
                }
              >
                Write this phrase down. You will want to store it in a secure location.
              </Text>
            </Component>
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                }
              }
            />
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                }
              }
            />
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                }
              }
            />
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                }
              }
            />
          </ScrollView>,
          <Component
            style={
              Object {
                "justifyContent": "flex-end",
              }
            }
          >
            <Button
              color="#4d9678"
              onPress={[Function]}
              title="I wrote it down"
            />
          </Component>,
        ],
        "style": Object {
          "backgroundColor": "#1c2227",
          "flex": 1,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Component>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 20,
                    }
                  }
                >
                  Seed phrase
                </Text>
              </Component>,
              <Component>
                <ProgressViewIOS
                  progress={0.625}
                />
              </Component>,
              <Component>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 20,
                    }
                  }
                >
                  Write this phrase down. You will want to store it in a secure location.
                </Text>
              </Component>,
              <Component
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                  }
                }
              />,
              <Component
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                  }
                }
              />,
              <Component
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                  }
                }
              />,
              <Component
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                  }
                }
              />,
            ],
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 20,
                    }
                  }
                >
                  Seed phrase
                </Text>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Seed phrase",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#ffffff",
                    "fontSize": 20,
                  },
                },
                "ref": null,
                "rendered": "Seed phrase",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <ProgressViewIOS
                  progress={0.625}
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "progress": 0.625,
                  "style": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 20,
                    }
                  }
                >
                  Write this phrase down. You will want to store it in a secure location.
                </Text>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Write this phrase down. You will want to store it in a secure location.",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#ffffff",
                    "fontSize": 20,
                  },
                },
                "ref": null,
                "rendered": "Write this phrase down. You will want to store it in a secure location.",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [],
                "style": Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                },
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [],
                "style": Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                },
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [],
                "style": Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                },
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [],
                "style": Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                },
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
          ],
          "type": [MockFunction],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Button
              color="#4d9678"
              onPress={[Function]}
              title="I wrote it down"
            />,
            "style": Object {
              "justifyContent": "flex-end",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "#4d9678",
              "onPress": [Function],
              "title": "I wrote it down",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Object {
              "backgroundColor": "#1c2227",
              "flex": 1,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
        >
          <ScrollView
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Component>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 20,
                  }
                }
              >
                Seed phrase
              </Text>
            </Component>
            <Component>
              <ProgressViewIOS
                progress={0.625}
              />
            </Component>
            <Component>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 20,
                  }
                }
              >
                Write this phrase down. You will want to store it in a secure location.
              </Text>
            </Component>
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                }
              }
            />
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                }
              }
            />
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                }
              }
            />
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-evenly",
                }
              }
            />
          </ScrollView>
          <Component
            style={
              Object {
                "justifyContent": "flex-end",
              }
            }
          >
            <Button
              color="#4d9678"
              onPress={[Function]}
              title="I wrote it down"
            />
          </Component>
        </Component>,
        "style": Object {
          "backgroundColor": "#1c2227",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <ScrollView
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Component>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 20,
                    }
                  }
                >
                  Seed phrase
                </Text>
              </Component>
              <Component>
                <ProgressViewIOS
                  progress={0.625}
                />
              </Component>
              <Component>
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#ffffff",
                      "fontSize": 20,
                    }
                  }
                >
                  Write this phrase down. You will want to store it in a secure location.
                </Text>
              </Component>
              <Component
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                  }
                }
              />
              <Component
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                  }
                }
              />
              <Component
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                  }
                }
              />
              <Component
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                  }
                }
              />
            </ScrollView>,
            <Component
              style={
                Object {
                  "justifyContent": "flex-end",
                }
              }
            >
              <Button
                color="#4d9678"
                onPress={[Function]}
                title="I wrote it down"
              />
            </Component>,
          ],
          "style": Object {
            "backgroundColor": "#1c2227",
            "flex": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Component>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Seed phrase
                  </Text>
                </Component>,
                <Component>
                  <ProgressViewIOS
                    progress={0.625}
                  />
                </Component>,
                <Component>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Write this phrase down. You will want to store it in a secure location.
                  </Text>
                </Component>,
                <Component
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-evenly",
                    }
                  }
                />,
                <Component
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-evenly",
                    }
                  }
                />,
                <Component
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-evenly",
                    }
                  }
                />,
                <Component
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-evenly",
                    }
                  }
                />,
              ],
              "style": Object {
                "flex": 1,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Seed phrase
                  </Text>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Seed phrase",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "color": "#ffffff",
                      "fontSize": 20,
                    },
                  },
                  "ref": null,
                  "rendered": "Seed phrase",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <ProgressViewIOS
                    progress={0.625}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "progress": 0.625,
                    "style": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Write this phrase down. You will want to store it in a secure location.
                  </Text>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": "Write this phrase down. You will want to store it in a secure location.",
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "color": "#ffffff",
                      "fontSize": 20,
                    },
                  },
                  "ref": null,
                  "rendered": "Write this phrase down. You will want to store it in a secure location.",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [],
                  "style": Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                  },
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [],
                  "style": Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                  },
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [],
                  "style": Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                  },
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [],
                  "style": Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-evenly",
                  },
                },
                "ref": null,
                "rendered": Array [],
                "type": [Function],
              },
            ],
            "type": [MockFunction],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Button
                color="#4d9678"
                onPress={[Function]}
                title="I wrote it down"
              />,
              "style": Object {
                "justifyContent": "flex-end",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "color": "#4d9678",
                "onPress": [Function],
                "title": "I wrote it down",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
